test_redis_client.go

```diff
@@ -126,7 +126,7 @@ func (client *TestRedisClient) LPush(key string, value ...string) (total int64,
 	}
 
 	client.storeList(key, append(value, list...))
-	return int64(len(list)) + 1, nil
+	return int64(len(list) + len(value)), nil
```

Comment:
I think the confusing part here was that `value` (singular) is actually a list of values. So let's probably rename that variable too. And then I think it gets a bit more readable by just giving that new list a name so we can return the length of that new list. Like this:
```go
// LPush inserts the specified values at the head of the list stored at key.
// If key does not exist, it is created as empty list before performing the push operations.
// When key holds a value that is not a list, an error is returned.
// It is possible to push multiple elements using a single command call just specifying multiple arguments
// at the end of the command. Elements are inserted one after the other to the head of the list,
// from the leftmost element to the rightmost element.
func (client *TestRedisClient) LPush(key string, values ...string) (total int64, err error) {
	lock.Lock()
	defer lock.Unlock()

	list, err := client.findList(key)
	if err != nil {
		return 0, nil
	}

	newList := append(values, list...)
	client.storeList(key, newList)
	return int64(len(newList)), nil
}
```
Oh, and now I see that `lock` is a static global variable. It should be a field of the `TestRedisClient` struct. Do you want to fix that too maybe?